src/videojs.ads.js

```diff
@@ -757,6 +757,12 @@ var
               } else {
                 this.state = 'ads-ready?';
               }
+              // When a new source is loaded into the player, we should remove the snapshot
+              // to avoid confusing player state with the new content's state
+              // i.e When new content is set, the player should fire the ended event
+              if (this.snapshot && this.snapshot.ended) {
```

Comment:
Essentially damage control. I saw differences in the event flow when I removed the snapshot regardless of whether it is ended or not.